### PR TITLE
Move cluster, webhub and store out of Server

### PR DESCRIPTION
**Pull request by isacikgoz**
#### Summary
This is the most important PR for migrating entities under the platform service. There are a few migrations included as:

- Cluster logic, and handlers -> platform
- Webhub -> platform
- Session cache and other caches -> platform
- Store -> platform
- License -> platform

So keeping cache in platform seemed to be a better idea to me as it's heavily coupled with clustering. Same for webhub, but there is a small problem with webhub that it also requires some upper level application logic. Therefore I added a `SuiteIFace` which essentially will be provided by the suite service. This allowed me to do some mocking as well.

I'd like to get a review on especially the platform and server start-up. As a result of the migration, I had to tweak the startup sequence which was already complicated. I had to reflect the `app.Option` to the platform to have a less impact on the sequence.

Ideally the service was going to boot with a configuration but it turns out that we need a dynamic configuration throughout the startup sequence hence using options made much more sense. In fact, as a next step we can completely remove startup config. Maybe it should be done after the merge to not complicate things even further.

Currently there are two things that bothers me:
Cluster implementation on enterprise requires `plugin.Environment`, I had to pass it to the Platform Service first from Channels then use it on the cluster, I think we need to address this in a better way. The latter complaint is Shared Channels Service, this also needs a further thinking.

#### Release Note

```release-note
NONE
```

### CODE REVIEW

File: app/platform/config.go [hunk 1]
```diff
@@ -33,6 +42,10 @@ func (c *ServiceConfig) validate() error {
 		return errors.New("ConfigStore is required")
 	}
 
+	// if c.Store == nil {
+	// 	return errors.New("Store is required")
+	// }
```
Comment (isacikgoz) [inline app/platform/config.go:47]: We will mandate after completing store migration.

**Review by agnivade — COMMENTED** [review]
I don't see the app/web_conn, app/web_hub code removed. Were you planning to do that later?

File: Makefile [hunk 2]
```diff
@@ -394,6 +394,10 @@ email-mocks: ## Creates mocks for misc interfaces.
 	$(GO) install github.com/vektra/mockery/v2/...@v2.10.4
 	$(GOBIN)/mockery --dir app/email --name ServiceInterface --output app/email/mocks --note 'Regenerate this file using `make email-mocks`.'
 
+platform-mocks: ## Creates mocks for platform interfaces.
+	$(GO) install github.com/vektra/mockery/v2/...@v2.14.0
+	$(GOBIN)/mockery --dir app/platform --name SuiteIFace --output app/platform/mocks --note 'Regenerate this file using `make platform-mocks`.'
```
Comment (agnivade) [inline Makefile:399]: Why is it called `SuiteIFace` when it generates platform mocks?

Comment (isacikgoz) [reply Makefile:399]: It's mock for suite interface, The future suite service will going to provide that interface.

File: Makefile [hunk 3]
```diff
@@ -399,7 +399,7 @@ platform-mocks: ## Creates mocks for platform interfaces.
 pluginapi: ## Generates api and hooks glue code for plugins
 	$(GO) generate $(GOFLAGS) ./plugin
 
-mocks: store-mocks telemetry-mocks filestore-mocks ldap-mocks plugin-mocks einterfaces-mocks searchengine-mocks sharedchannel-mocks misc-mocks email-mocks
+mocks: store-mocks telemetry-mocks filestore-mocks ldap-mocks plugin-mocks einterfaces-mocks searchengine-mocks sharedchannel-mocks misc-mocks email-mocks platform-mocks
```
Comment (noxer) [inline Makefile:402]: ❤️

File: Makefile (hunk 2 above)
Comment (agnivade) [reply Makefile:397]: Ah thanks

**Comment by isacikgoz** [comment]
@agnivade @noxer Let me address the issues with the tests before you start reviewing. It seems like there are some substantial errors.

**Review by agnivade — COMMENTED** [review]
Finished 40/213 for now. Will continue the rest tomorrow. :P

**Review by agnivade — COMMENTED** [review]
100 files done! Frankly I don't think I can give this a thorough review. It's too large to do a line by line review. Let's just make the CI green and start e2e tests.

**Comment by agnivade** [comment]
@saturninoabril - This is an important and big PR that we would like to have some extensive testing before merging. 

Some of the things we'd like to ensure:
1. Run this PR with e2e tests in HA mode and ensure no issues.
2. Do some smoke tests in HA mode and ensure no issues related to passing ws messages.

Would you be able to nominate someone from your team to look into this?

**Comment by saturninoabril** [comment]
Thanks @lindalumitchell for gladly accepting to review this PR!

**Comment by agnivade** [comment]
One more thing: Plugins! Test with plugins running on both nodes and ensure things are fine.

**Comment by agnivade** [comment]
@lindalumitchell - circling back on this again. Did you get a chance to test this out? It is accumulating merge conflicts, so let's try to prioritize this.

**Comment by lindalumitchell** [comment]
@agnivade apologies for the delay and thanks for the mention. Will do. @saturninoabril did you finish all you wanted to do E2E-wise with this one?

**Review by lindalumitchell — APPROVED** [review]
Ah Saturn is out of office at the moment. I looked through our discussions and believe that he did complete the E2Es he wanted to do. I ran through the results of one run and it looked okay.

Smoke tests throughout also look good; no issues found.

**Review by agnivade — APPROVED** [review]
Finally :)

**Comment by agnivade** [comment]
> Cluster implementation on enterprise requires plugin.Environment, I had to pass it to the Platform Service first from Channels then use it on the cluster, I think we need to address this in a better way. The latter complaint is Shared Channels Service, this also needs a further thinking.

Agree. But I think we need to start prioritizing other OKRs now. It is clear that any sort of major refactoring like this takes a lot of time and effort from multiple teams. As discussed on our call, let's try to wrap things up by this week or the next at most. I and Tim can help out with the other small things pending.

**Comment by isacikgoz** [comment]
ping @noxer

**Review by noxer — APPROVED** [review]
LGTM, sorry for the delay. It's a lot of files :)

**Pull request merged**